Framework/Kernel/inc/MantidKernel/Cache.h

```diff
@@ -74,7 +99,26 @@ template <class KEYTYPE, class VALUETYPE> class DLLExport Cache {
    * @param value The new value for the key
    */
   void setCache(const KEYTYPE &key, const VALUETYPE &value) {
-    m_cacheMap[key] = value;
+    //
+    auto where = m_cacheMap.find(key);
```

Comment:
The [documentation](https://software.intel.com/en-us/node/506171) claims concurrent access to `find` and `insert` is allowed, but not `unsafe_erase`.